libs/sdk-ui-ext/src/internal/components/configurationControls/comparison/label/LabelSubSection.tsx

```diff
@@ -4,54 +4,165 @@ import { useIntl } from "react-intl";
 import InputControl from "../../InputControl.js";
-import { COMPARISON_LABEL_UNCONDITIONAL_VALUE_PATH } from "../ComparisonValuePath.js";
-import { ComparisonPositionValues } from "@gooddata/sdk-ui-charts";
+import {
+    COMPARISON_LABEL_CONDITIONAL_ENABLED_VALUE_PATH,
+    COMPARISON_LABEL_EQUALS_VALUE_PATH,
+    COMPARISON_LABEL_NEGATIVE_VALUE_PATH,
+    COMPARISON_LABEL_POSITIVE_VALUE_PATH,
+    COMPARISON_LABEL_UNCONDITIONAL_VALUE_PATH,
+} from "../ComparisonValuePath.js";
+import {
+    ComparisonPositionValues,
+    CalculateAs,
+    IDefaultLabelKeys,
+    CalculationType,
+} from "@gooddata/sdk-ui-charts";
+import CheckboxControl from "../../CheckboxControl.js";
 
 export interface ILabelSubSectionProps {
     sectionDisabled: boolean;
     showDisabledMessage?: boolean;
-    defaultLabelKey: string;
+    defaultLabelKeys: IDefaultLabelKeys;
+    calculationType: CalculationType;
     properties: IVisualizationProperties<IComparisonControlProperties>;
     pushData: PushDataCallback;
 }
 
+const DISABLED_MESSAGE_ALIGN_POINTS = [{ align: "cr cl", offset: { x: 0, y: 7 } }];
+
+function getDisabledMessageId(showDisabledMessage: boolean, isPositionOnTop: boolean, isCalculateAsRatio: boolean): string {
+    if (showDisabledMessage) {
+        return messages.notApplicable.id;
+    }
+    if (isPositionOnTop) {
+        return comparisonMessages.labelPositionOnTopDisabled.id;
+    }
+    if (isCalculateAsRatio) {
+        return comparisonMessages.labelConditionalDisabledByRatio.id;
+    }
+}
+
 const LabelSubSection: React.FC<ILabelSubSectionProps> = ({
     sectionDisabled,
     showDisabledMessage,
-    defaultLabelKey,
+    defaultLabelKeys,
+    calculationType,
     properties,
     pushData,
 }) => {
     const { formatMessage } = useIntl();
-
+    const isCalculateAsRatio = calculationType === CalculateAs.RATIO;
     const isPositionOnTop = properties?.controls?.comparison?.position === ComparisonPositionValues.TOP;
     const disabled = sectionDisabled || isPositionOnTop;
     const shouldShowDisabledMessage = showDisabledMessage || isPositionOnTop;
-    const disabledMessageId = isPositionOnTop ? comparisonMessages.labelPositionOnTopDisabled.id : undefined;
+
+    const conditionalDisabled = disabled || isCalculateAsRatio;
+    const shouldShowConditionalDisabledMessage = shouldShowDisabledMessage || isCalculateAsRatio;
+    const disabledMessageId = getDisabledMessageId(showDisabledMessage, isPositionOnTop, isCalculateAsRatio);
 
     const defaultValue =
         properties?.controls?.comparison?.labelConfig?.unconditionalValue ||
-        formatMessage({ id: defaultLabelKey });
+        formatMessage({ id: defaultLabelKeys.nonConditionalKey });
 
+    const conditionalEnabled =
+        !!properties?.controls?.comparison?.labelConfig?.isConditional &&
+        calculationType !== CalculateAs.RATIO;
+    let positiveDefaultValue, negativeDefaultValue, equalsDefaultValue;
+    if (conditionalEnabled) {
```

Comment:
consider to remove the condition and change the variable name:
defaultValue -> unconditionalValue
positiveDefaultValue -> positiveValue
negativeDefaultValue -> negativeValue
equalsDefaultValue -> equalsValue

Reply:
The condition is necessary, these values should not be computed if users do not enable the conditional labels.
BTW, I renamed as your suggestion. Thank you.